### PR TITLE
Add doc on how to develop new device plugins

### PR DESCRIPTION
**Pull request by rojkov**

### CODE REVIEW

**Comment by rojkov** [comment]
@MCamp859 may I ask you to review this update?

File: DEVEL.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                },
+            },
+        })
+        ...
+        notifier.Notify(devTree)
+    }
+}
+```
+
+Optionally your device plugin may implement the `deviceplugin.PostAllocator`
```
Comment (MCamp859) [inline DEVEL.md]: Optionally, your device plugin may also implement the `deviceplugin.PostAllocator`
(Add "also". Explain why you would want to do this, does it improve the plugin in some way?)

File: DEVEL.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+interface. If implemented its method `PostAllocate()` modifies
+`pluginapi.AllocateResponse` responses just before they are sent to `kubelet`.
```
Comment (MCamp859) [inline DEVEL.md]: Suggest adding a new line at the end to point to additional info upstream (if it exists). 
For more information, refer to [\upstream topic name]\(\URL)

**Comment by rojkov** [comment]
@MCamp859 thank you! I've updated the PR.

This doc explains how to use our library, not upstream functionality. But I feel reluctant to say it's a cool thing, because when everything is "cool" the word deflates its value and my first reaction usually is "another bait, proceed to the next thing". :)

**Comment by MCamp859** [comment]
LGTM, thanks!

**Pull request merged**